skimage/color/colorconv.py

```diff
@@ -780,6 +783,9 @@ def rgb2gray(rgb):
 
     If there is an alpha channel present, it is ignored.
 
+    If the input array is 2-D, it is converted to a row-major (i.e. C-style)
```

Comment:
I would be fine adding al lthe comments in the `Notes` section stating that calling this function with `2D` images, while supported for now, is heavily discouraged.